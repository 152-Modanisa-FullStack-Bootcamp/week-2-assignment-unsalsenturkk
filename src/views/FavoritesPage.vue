<template>
  <div
      id="favoritespage"
      class="m-flex"
  >
    <div
        id="favorite-contents"
        class="m-flex
               m-flex-column"
    >
      <FavoriteContent
          v-for="video in filteredFavoriVideos"
          :key="video.id"
          :video="video"
      >
      </FavoriteContent>
    </div>
  </div>
</template>

<script>
import FavoriteContent from "@/components/FavoriteContent";
import API from "@/api";
import {mapGetters} from "vuex"

export default {
  name: "FavoritesPage",
  data() {
    return {
      videos: []
    }
  },
  components: {
    FavoriteContent
  },
  async mounted() {
    this.videos = await API.getVideoList();
  },
  computed: {
    ...mapGetters({
      getFavoriteVideos: "getFavoriteVideos",
      textSearch: 'getSearchText'
    }),
    filteredFavoriVideos() {
      return this.getFavoriteVideos.filter(
          video => video.title.includes(this.textSearch)
      );
    }

  },


}
</script>

<style scoped>
#favoritespage {
  width: 50%;
  height: auto;
}

#favorite-contents {
  width: 100%;
}
</style>