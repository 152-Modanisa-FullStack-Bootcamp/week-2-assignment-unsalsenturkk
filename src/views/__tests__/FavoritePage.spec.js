import FavoritesPage from "@/views/FavoritesPage";
import {createLocalVue, shallowMount} from "@vue/test-utils";
import API from "@/api";
import Vuex from "vuex";
import flushPromises from "flush-promises";

jest.mock("@/api")

describe("FavoritesPage.vue",()=>{
    it('sanity test',  () => {
        return
    });

    describe("FavoritesPage unit tests",()=>{

        it('should FavoritePage exists',  async () => {
            const mockData = {
                "id": 1,
                "videoAddress": "https://www.youtube.com/watch?v=FXpIoQ_rT_c",
                "coverImage": "https://raw.githubusercontent.com/modanisa/bootcamp-video-db/main/video-images/1-cover.webp",
                "hoverImage": "https://raw.githubusercontent.com/modanisa/bootcamp-video-db/main/video-images/1-hover.webp",
                "title": "Vue.js Course for Beginners [2021 Tutorial]",
                "viewCount": 254,
                "publishDateInMonth": 4,
                "ownerImage": "https://yt3.ggpht.com/ytc/AKedOLTtJvQ1Vfew91vemeLaLdhjOwGx3tTBLlreK_QUyA=s68-c-k-c0x00ffffff-no-rj",
                "ownerName": "freeCodeCamp.org",
                "description": "Learn Vue 3 by in this full course. Vue.js is an open-source model–view–view model front end JavaScript framework for building user interfaces and single-page applications."
            }
            const mockResponse = [
                mockData
            ]
            API.getVideoList.mockResolvedValue(mockResponse)

            const localVue = createLocalVue()
            localVue.use(Vuex)

            const getters = {
                getSearchText : () => "",
                getFavoriteVideos : () => [mockData]
            }


            const store = new Vuex.Store({
                getters
            })

            const wrapper = shallowMount(FavoritesPage,{
                localVue,
                store
            })

            await flushPromises()
            expect(wrapper.exists()).toBeTruthy()
        });
    })

    describe("FavoritesPage snapshot test",()=>{
        it('should renders FavoritePage correclty',  async () => {
            const mockData = {
                "id": 1,
                "videoAddress": "https://www.youtube.com/watch?v=FXpIoQ_rT_c",
                "coverImage": "https://raw.githubusercontent.com/modanisa/bootcamp-video-db/main/video-images/1-cover.webp",
                "hoverImage": "https://raw.githubusercontent.com/modanisa/bootcamp-video-db/main/video-images/1-hover.webp",
                "title": "Vue.js Course for Beginners [2021 Tutorial]",
                "viewCount": 254,
                "publishDateInMonth": 4,
                "ownerImage": "https://yt3.ggpht.com/ytc/AKedOLTtJvQ1Vfew91vemeLaLdhjOwGx3tTBLlreK_QUyA=s68-c-k-c0x00ffffff-no-rj",
                "ownerName": "freeCodeCamp.org",
                "description": "Learn Vue 3 by in this full course. Vue.js is an open-source model–view–view model front end JavaScript framework for building user interfaces and single-page applications."
            }
            const mockResponse = [
                mockData
            ]
            API.getVideoList.mockResolvedValue(mockResponse)

            const localVue = createLocalVue()
            localVue.use(Vuex)

            const getters = {
                getSearchText : () => "",
                getFavoriteVideos : () => [mockData]
            }


            const store = new Vuex.Store({
                getters
            })

            const wrapper = shallowMount(FavoritesPage,{
                localVue,
                store
            })

            await flushPromises()
            expect(wrapper.element).toMatchSnapshot()
        });
    })
})