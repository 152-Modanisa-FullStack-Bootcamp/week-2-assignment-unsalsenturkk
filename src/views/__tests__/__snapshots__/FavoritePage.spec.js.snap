// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FavoritesPage.vue FavoritesPage snapshot test should renders FavoritePage correclty 1`] = `
<div
  class="m-flex"
  id="favoritespage"
>
  <div
    class="m-flex
             m-flex-column"
    id="favorite-contents"
  >
    <favoritecontent-stub
      video="[object Object]"
    />
  </div>
</div>
`;
